Build failure: gittyup

## Steps To Reproduce

Steps to reproduce the behavior:

1. build *gittyup*

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
$ nix shell nixpkgs#gittyup
warning: download buffer is full; consider increasing the 'download-buffer-size' setting
error: builder for '/nix/store/zvwg1ms2y5hlj0yhpi1brwnrs1jlpk36-gittyup-1.4.0.drv' failed with exit code 1;
       last 25 log lines:
       > [399/577] Linking CXX executable src/index/index_test
       > [400/577] Building C object test/dep/zip/CMakeFiles/zip.dir/src/zip.c.o
       > FAILED: test/dep/zip/CMakeFiles/zip.dir/src/zip.c.o
       > /nix/store/xcn9p4xxfbvlkpah7pwchpav4ab9d135-gcc-wrapper-14-20241116/bin/gcc -DCURR_YEAR=1980 -DDEBUG_OUTPUT_GENERAL -I/build/source/test/dep/zip/src -Wall -Wextra -Werror -pedantic -O3 -DNDEBUG -fPIC -fvisibility=hidden -MD -MT test/dep/zip/CMakeFiles/zip.dir/src/zip.c.o -MF test/dep/zip/CMakeFiles/zip.dir/src/zip.c.o.d -o test/dep/zip/CMakeFiles/zip.dir/src/zip.c.o -c /build/source/test/dep/zip/src/zip.c
       > In file included from /build/source/test/dep/zip/src/zip.c:39:
       > /build/source/test/dep/zip/src/miniz.h:4988:9: note: '#pragma message: Using fopen, ftello, fseeko, stat() etc. path for file I/O - this path may not support large files.'
       >  4988 | #pragma message(                                                               \
       >       |         ^~~~~~~
       > /build/source/test/dep/zip/src/zip.c: In function 'zip_stream_copy':
       > /build/source/test/dep/zip/src/zip.c:1676:24: error: 'calloc' sizes specified with 'sizeof' in the earlier argument and not in the later argument [-Werror=calloc-transposed-args]
       >  1676 |   *buf = calloc(sizeof(unsigned char), n);
       >       |                        ^~~~~~~~
       > /build/source/test/dep/zip/src/zip.c:1676:24: note: earlier argument should specify number of elements, later size of each element
       > cc1: all warnings being treated as errors
       > [401/577] Automatic MOC for target ui
       > [402/577] Building CXX object src/index/CMakeFiles/indexer.dir/indexer.cpp.o
       > /build/source/src/index/indexer.cpp: In function 'void {anonymous}::term(int)':
       > /build/source/src/index/indexer.cpp:100:8: warning: ignoring return value of 'ssize_t write(int, const void*, size_t)' declared with attribute 'warn_unused_result' [-Wunused-result]
       >   100 |   write(fds[0], &ch, sizeof(ch));
       >       |   ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
       > /build/source/src/index/indexer.cpp: In lambda function:
       > /build/source/src/index/indexer.cpp:408:13: warning: ignoring return value of 'ssize_t read(int, void*, size_t)' declared with attribute 'warn_unused_result' [-Wunused-result]
       >   408 |         read(fds[1], &ch, sizeof(ch));
       >       |         ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/zvwg1ms2y5hlj0yhpi1brwnrs1jlpk36-gittyup-1.4.0.drv'.```
```
</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.8, NixOS, 25.05 (Warbler), 25.05.20250104.8f3e1f8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/khbvilmsrv4l69nwd52h27j1mp44a0xi-source`

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

---
CC: @r-ryantm
(The `maintainers` array of this package is empty, so I did a `git blame`, and @r-ryantm bumped this package a few months ago, so maybe they can help. Thanks!

Note for maintainers: Please tag this issue in your PR.
---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/version-management/gittyup/0001-Fix-incorrect-order-of-argument-to-calloc-345.patch
+++ b/pkgs/applications/version-management/gittyup/0001-Fix-incorrect-order-of-argument-to-calloc-345.patch
@@ -1,0 +1,25 @@
+From 687b5424128fd637555be4a0d18c72c4e870bd6f Mon Sep 17 00:00:00 2001
+From: Alfie <30699769+AHSauge@users.noreply.github.com>
+Date: Sat, 11 May 2024 13:11:08 +0200
+Subject: [PATCH] Fix incorrect order of argument to calloc (#345)
+
+---
+ src/zip.c | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/src/zip.c b/src/zip.c
+index b14f9e3..a7e5ef8 100644
+--- a/test/dep/zip/src/zip.c
++++ b/test/dep/zip/src/zip.c
+@@ -1874,7 +1874,7 @@ ssize_t zip_stream_copy(struct zip_t *zip, void **buf, size_t *bufsize) {
+     *bufsize = n;
+   }
+ 
+-  *buf = calloc(sizeof(unsigned char), n);
++  *buf = calloc(n, sizeof(unsigned char));
+   memcpy(*buf, zip->archive.m_pState->m_pMem, n);
+ 
+   return (ssize_t)n;
+-- 
+2.47.0
+

--- a/pkgs/applications/version-management/gittyup/default.nix
+++ b/pkgs/applications/version-management/gittyup/default.nix
@@ -48,6 +48,7 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [
     cmake
+    cmark
     ninja
     pkg-config
     wrapQtAppsHook

--- a/pkgs/applications/version-management/gittyup/default.nix
+++ b/pkgs/applications/version-management/gittyup/default.nix
@@ -29,6 +29,12 @@ stdenv.mkDerivation rec {
     fetchSubmodules = true;
   };
 
+  patches = [
+    # Fix GCC 14 build error (remove for next update)
+    # https://github.com/Murmele/Gittyup/pull/759
+    ./0001-Fix-incorrect-order-of-argument-to-calloc-345.patch
+  ];
+
   cmakeFlags =
     let
       inherit (lib) cmakeBool;

--- a/pkgs/applications/version-management/gittyup/default.nix
+++ b/pkgs/applications/version-management/gittyup/default.nix
@@ -88,7 +88,10 @@ stdenv.mkDerivation rec {
     description = "Graphical Git client designed to help you understand and manage your source code history";
     homepage = "https://murmele.github.io/Gittyup";
     license = with licenses; [ mit ];
-    maintainers = [ ];
+    maintainers = with maintainers; [
+      fliegendewurst
+      phijor
+    ];
     platforms = platforms.unix;
     broken = stdenv.hostPlatform.isDarwin;
   };

--- a/pkgs/applications/version-management/gittyup/default.nix
+++ b/pkgs/applications/version-management/gittyup/default.nix
@@ -29,18 +29,22 @@ stdenv.mkDerivation rec {
     fetchSubmodules = true;
   };
 
-  cmakeFlags = [
-    "-DBUILD_SHARED_LIBS=OFF"
-    "-DUSE_SYSTEM_CMARK=ON"
-    "-DUSE_SYSTEM_GIT=ON"
-    "-DUSE_SYSTEM_HUNSPELL=ON"
-    # upstream uses its own fork of libgit2 as of 1.2.2, however this may change in the future
-    # "-DUSE_SYSTEM_LIBGIT2=ON"
-    "-DUSE_SYSTEM_LIBSSH2=ON"
-    "-DUSE_SYSTEM_LUA=ON"
-    "-DUSE_SYSTEM_OPENSSL=ON"
-    "-DENABLE_UPDATE_OVER_GUI=OFF"
-  ];
+  cmakeFlags =
+    let
+      inherit (lib) cmakeBool;
+    in
+    [
+      (cmakeBool "BUILD_SHARED_LIBS" false)
+      (cmakeBool "USE_SYSTEM_CMARK" true)
+      (cmakeBool "USE_SYSTEM_GIT" true)
+      (cmakeBool "USE_SYSTEM_HUNSPELL" true)
+      # upstream uses its own fork of libgit2 as of 1.2.2, however this may change in the future
+      # (cmakeBool "USE_SYSTEM_LIBGIT2" true)
+      (cmakeBool "USE_SYSTEM_LIBSSH2" true)
+      (cmakeBool "USE_SYSTEM_LUA" true)
+      (cmakeBool "USE_SYSTEM_OPENSSL" true)
+      (cmakeBool "ENABLE_UPDATE_OVER_GUI" false)
+    ];
 
   nativeBuildInputs = [
     cmake

